test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetRestoreTests.cs

```diff
@@ -1413,6 +1413,72 @@ public async Task DotnetRestore_MultiTargettingProject_WithDifferentProjectRefer
             }
         }
 
+        [Fact]
+        public void CollectFrameworkReferences_WithTransitiveFrameworkReferences_ExcludesTransitiveFrameworkReferences()
+        {
+            using (var pathContext = _msbuildFixture.CreateSimpleTestPathContext())
+            {
+                // Arrange
+                // Library project with Framework Reference
+                var libraryName = "Library";
+                var libraryProjectFilePath = Path.Combine(Path.Combine(pathContext.SolutionRoot, libraryName), $"{libraryName}.csproj");
+                _msbuildFixture.CreateDotnetNewProject(pathContext.SolutionRoot, libraryName, "classlib");
+
+                using (var stream = File.Open(libraryProjectFilePath, FileMode.Open, FileAccess.ReadWrite))
+                {
+                    var xml = XDocument.Load(stream);
+                    ProjectFileUtils.AddItem(
+                        xml,
+                        name: "FrameworkReference",
+                        identity: "Microsoft.AspNetCore.App",
+                        framework: (string)null,
+                        attributes: new Dictionary<string, string>(),
+                        properties: new Dictionary<string, string>()
+                        );
+                    ProjectFileUtils.WriteXmlToFile(xml, stream);
+                }
+
+                var packResult = _msbuildFixture.RunDotnet(pathContext.SolutionRoot, $"pack {libraryProjectFilePath} /p:PackageOutputPath=\"{pathContext.PackageSource}\"", ignoreExitCode: true);
+                packResult.Success.Should().BeTrue(because: packResult.AllOutput);
+
+                // Consumer project.
+                var consumerName = "Consumer";
+                _msbuildFixture.CreateDotnetNewProject(pathContext.SolutionRoot, consumerName, "console");
+
+                var consumerProjectFilePath = Path.Combine(Path.Combine(pathContext.SolutionRoot, consumerName), $"{consumerName}.csproj");
+
+                using (var stream = File.Open(consumerProjectFilePath, FileMode.Open, FileAccess.ReadWrite))
+                {
+                    var xml = XDocument.Load(stream);
+                    ProjectFileUtils.AddItem(
+                        xml,
+                        name: "PackageReference",
+                        identity: "Library",
+                        framework: (string)null,
+                        attributes: new Dictionary<string, string>() { { "Version", "1.0.0" } },
+                        properties: new Dictionary<string, string>()
+                        );
+                    ProjectFileUtils.WriteXmlToFile(xml, stream);
+                }
+
+                // Add a DBT.
+
+                var directoryBuildTargetsFilePath = Path.Combine(pathContext.SolutionRoot, "Directory.Build.targets");
```

Comment:
Personally, I find the code sufficiently self-explanatory. Actually, DBT confused me, as I've never seen that before, so I was going to suggest expanding the initialism, but then I realized the comment is just explaining what the code is doing, not why.

```suggestion
                var directoryBuildTargetsFilePath = Path.Combine(pathContext.SolutionRoot, "Directory.Build.targets");
```